app/models/attachment.rb

```diff
@@ -2,11 +2,22 @@
 
 # entity class for Attachment
 class Attachment < ApplicationRecord
+  include HasPuid
+
+  FORMAT_REGEX = {
+    fasta: /^\S+\.fn?a(sta)?(\.gz)?$/,
+    fastq: /^\S+\.f(ast)?q(\.gz)?$/,
+    text: /^\S+\.(txt|rtf)?(\.gz)?$/,
+    csv: /^\S+\.(csv)?(\.gz)?$/,
+    tsv: /^\S+\.(tsv)?(\.gz)?$/,
+    spreadsheet: /^\S+\.(xls|xlsx)?$/,
+    json: /^\S+\.(json)?(\.gz)?$/,
+    genbank: /^\S+\.(gbk|gbf|genbank)?(\.gz)?$/
+  }.freeze
```

Comment:
Because when you assign these to the `metadata`, you're switching them to strings with `.to_s`, and I would need to do the same in my PR, could you change the hash keys to use strings. Ie: `{'key' => 'value'}, that way you do not need to use `.to_s` below.